src/chatClient.cpp

```diff
@@ -2107,23 +2144,30 @@ IApp::IPeerChatListItem* PeerChatRoom::addAppItem()
     return list ? list->addPeerChatItem(*this) : nullptr;
 }
 
+//Resume from cache
 PeerChatRoom::PeerChatRoom(ChatRoomList& parent, const uint64_t& chatid,
-    unsigned char aShard, chatd::Priv aOwnPriv, const uint64_t& peer, chatd::Priv peerPriv, int64_t ts, bool aIsArchived)
-:ChatRoom(parent, chatid, false, aShard, aOwnPriv, ts, aIsArchived), mPeer(peer),
-  mPeerPriv(peerPriv),
-  mRoomGui(nullptr)
+    unsigned char aShard, chatd::Priv aOwnPriv, const uint64_t& peer,
+    chatd::Priv peerPriv, int64_t ts, bool aIsArchived, const std::string& url)
+    :ChatRoom(parent, chatid, false, aShard, aOwnPriv, ts, aIsArchived),
+    mPeer(peer),
+    mPeerPriv(peerPriv),
+    mRoomGui(nullptr)
 {
+    mUrl = url;
```

Comment:
Better as part of the initialization list :)
Also, pass the `url` to the `ChatRoom` ctor, since it's its member.